web-api-db-ctrl/postgres/tenant/20230713173212_document.sql

```diff
@@ -0,0 +1,29 @@
+-- Copyright 2023 Xayn AG
+--
+-- This program is free software: you can redistribute it and/or modify
+-- it under the terms of the GNU Affero General Public License as
+-- published by the Free Software Foundation, version 3.
+--
+-- This program is distributed in the hope that it will be useful,
+-- but WITHOUT ANY WARRANTY; without even the implied warranty of
+-- MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+-- GNU Affero General Public License for more details.
+--
+-- You should have received a copy of the GNU Affero General Public License
+-- along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+CREATE TABLE snippet (
+    document_id TEXT NOT NULL
+        REFERENCES document(document_id) ON DELETE CASCADE,
+    snippet_idx INTEGER NOT NULL,
+    snippet TEXT NOT NULL,
+    embedding FLOAT4[] NOT NULL,
+
+    PRIMARY KEY (document_id, snippet_idx)
+);
```

Comment:
I'm quite split between having a `snippets TEXT[]` + `embedding jsonb[]` fields in document instead of another table as in general we always add/replace/delete all snippets when adding/replacing/deleting a document.

I went for this here as it allows us to only fetch the snippets we need instead of always returning all snippets, which can be potentially very many with embedding being potentially >1KiB per embedding.

Reply:
i would agree that the benefit of faster fetching outweighs the cognitive load coming with multiple tables.